### PR TITLE
Ppp autotools

### PR DESCRIPTION
**Pull request by enaess**
Fix for issue #339

This allows a user to configure the lock directory via configure variables.

### CODE REVIEW

**Comment by enaess** [comment]
Just a note here. I am not thrilled that the PPP project introduces new variable names like _PATH_XYZ, the <paths.h> from the system already defines e.g. _PATH_VARLOG and others. Changes are that if a new path was added globally with the same name, we'd have to #undef it; technically I kind of wanted to change this to PPP_PATH_XYZ instead (or something similar) to explicitly state the variable came from the PPP project.

**Comment by wongsyrone** [comment]
I think the `PPP_` prefix is more appropriate.

**Comment by thesamesam** [comment]
> Just a note here. I am not thrilled that the PPP project introduces new variable names like _PATH_XYZ, the <paths.h> from the system already defines e.g. _PATH_VARLOG and others. Changes are that if a new path was added globally with the same name, we'd have to #undef it; technically I kind of wanted to change this to PPP_PATH_XYZ instead (or something similar) to explicitly state the variable came from the PPP project. Again, change here was based on the minimum amount of work needed. Let me know if this suffice, or a more extensive fix is needed.

Yeah, no more reserved names please: https://www.gnu.org/software/libc/manual/html_node/Reserved-Names.html. It's far too fragile unfortunately.

**Review by thesamesam — CHANGES_REQUESTED** [review]
lgtm once the reserved names are fixed

**Comment by enaess** [comment]
OK, fixed up the path variables with a PPP_ prefix. Waiting for the CI checks to pass.

File: pppd/sha1.c
Comment (wongsyrone) [inline pppd/sha1.c:26]: Why not put this line in sha1.h

Comment (wongsyrone) [reply pppd/sha1.c:26]: There are symbols cannot be found like https://github.com/ppp-project/ppp/blob/c0d579b/pppd/sha1.h#L25

**Comment by enaess** [comment]
What is the error output you get when you compile?



Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

File: pppd/sha1.c
Comment (enaess) [reply pppd/sha1.c]: I've dropped the change from the patch-set. Seems like the best way is to try to document what compile issue you are seeing and let us figure out the best solution. Please file a bug as @thesamesam suggested.

Generally, I am not a fan of putting includes in header files.

**Comment by enaess** [comment]
Thanks @paulusmack

**Pull request merged**
